### PR TITLE
Fix download snapshot warnings

### PR DESCRIPTION
**Pull request by ammachado**
## What's changed?
Fix download warnings for SNAPSHOT components

## What's your motivation?
- Fix https://github.com/openrewrite/rewrite/issues/4508

## Anything in particular you'd like reviewers to focus on?
No

## Anyone you would like to review specifically?
@timtebeek 

## Have you considered any alternatives or workarounds?
No

## Any additional context
No.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java [hunk 1]
```diff
@@ -800,4 +804,22 @@ private void logDebug(MavenProject mavenProject, String message) {
...
+                project.getVersion().endsWith("-SNAPSHOT") ? null : project.getVersion()
+        );
+    }
+
+    private static Pom createPom(ResolvedGroupArtifactVersion gav, MavenProject project) {
+        Pom.PomBuilder builder = Pom.builder()
+                .gav(gav)
+                ;
+        // TODO: complete mapping
+        return builder.build();
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java:823]: I guess this is why it's still a WIP; let me know your plans and whether you'd need any help.

**Review by timtebeek — COMMENTED** [review]
Great start! Thanks for the various improvements throughout and the cache introduction to fix the underlying issue. Let me know once you've worked through the final todo before a last review & merge.

File: src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java [hunk 2]
```diff
@@ -807,4 +816,23 @@ private void logDebug(MavenProject mavenProject, String message) {
...
+                project.getArtifactId(),
+                project.getVersion(),
+                project.getVersion().endsWith("-SNAPSHOT") ? null : project.getVersion()
+        );
+    }
+
+    private static @Nullable Pom createPom(ResolvedGroupArtifactVersion gav, MavenProject project) {
+        try (InputStream is = Files.newInputStream(project.getFile().toPath())) {
+            RawPom rawPom = RawPom.parse(is, null);
+            return rawPom.toPom(project.getBasedir().toPath(), null);
```
Comment (ammachado) [inline src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java:833]: Should we use `ctx.getLocalRepository().getUri()` here too?

**Review by timtebeek — COMMENTED** [review]
Great to see this @ammachado ; it was a bit of a puzzle for me correlating this with the MavenParser in openrewrite/rewrite, but I came up with a couple of questions that would be good to address I think. Curious to hear your thoughts, as you've perhaps already looked into those aspects.

I've also added a minimal test since I noticed none were using snapshot versions before. We might want to extend that with some assertions, although that might be a bit of a challenge with the testing framework used for plugins.

File: src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java (hunk 2 above)
Comment (ammachado) [reply src/main/java/org/openrewrite/maven/MavenMojoProjectParser.java:830]: I copied the idea from `MavenParser`.

**Review by timtebeek — APPROVED** [review]
Thanks again @ammachado ! Great to see these cleared out.
